src/Uno.UI/UI/Xaml/FrameworkTemplate.cs

```diff
@@ -84,23 +92,34 @@ public FrameworkTemplate(object? owner, FrameworkTemplateBuilder? factory)
 		/// Creates a new instance of the current template.
 		/// </summary>
 		/// <returns>A new instance of the template</returns>
-		View? IFrameworkTemplateInternal.LoadContent()
+		View? IFrameworkTemplateInternal.LoadContent(DependencyObject? templatedParent)
+		//View? IFrameworkTemplateInternal.LoadContent(TemplateMaterializationSettings settings)
 		{
-			View? view = null;
 			try
 			{
 				ResourceResolver.PushNewScope(_xamlScope);
-				if (_viewFactory != null)
+
+				var members = new List<DependencyObject>();
+				var settings = new TemplateMaterializationSettings(templatedParent, members.Add);
+
+				var view = _viewFactory?.Invoke(_ownerRef?.Target, settings);
+
+				if (view is { })
 				{
-					view = _viewFactory(_ownerRef?.Target);
+					FrameworkTemplatePool.Instance.TrackMaterializedTemplate(this, view, members);
```

Comment:
we can optimize the materialized members tracking part 
by putting them into #if INSERT_MEANINGFUL_NAME

Reply:
do we have the other sides implemented?:
- xamlreader
- generated xaml code